deploy-android-app/action.yml

```
@@ -0,0 +1,193 @@
+    required: true
+  alias:
+    description: An alias used for the keystore file
+    required: true
+  keystore_password:
+    description: The password used for the keystore
+    required: true
+  alias_password:
+    description: The password used for the alias
+    required: true
+  service_account_json:
+    description: The JSON file for the service account
+    required: true
+  package_name:
+    description: The package name of the Android application being uploaded to the Play Store
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+
+    #Checkout Code
+    - name: Checkout Code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        token: ${{ inputs.github_token }}
+
+    #Validate Inputs
+    - name: Inputs validation
+      shell: bash
+      run: |
+        [[ "${{ inputs.github_token }}" ]] || { echo "github_token input is empty" ; exit 1; }
+
+    #SETUP GIT LOCAL USER
+    - name: Setup Git
+      shell: bash
+      run: |
+        git config user.name "${{ inputs.who_trigger }}"
+        git config user.email "${{ inputs.who_trigger }}"
+      env:
+        GITHUB_TOKEN: ${{ inputs.github_token }}
+
+    #Increase Version Name NON-PATCH
+    - name: Bump version major-minor
+      uses: HardNorth/github-version-generate@v1.4.0
+      if: inputs.version != 'patch'
+      with:
+        version-source: file
+        version-file: VERSION
+        release-version-cut-prerelease: true
+        next-version-increment-patch: true
+        next-version-increment-minor: ${{ contains(inputs.version, 'minor') }}
+        next-version-increment-major: ${{ contains(inputs.version, 'major') }}
+
+    #UPDATE Version Name NON-PATCH
+    - name: Update version
+      shell: bash
+      if: inputs.version != 'patch'
+      run: |
+        echo "${{ env.NEXT_VERSION }}" > VERSION
+
+    #Increase Version Name PATCH
+    - name: Generate versions
+      uses: HardNorth/github-version-generate@v1.4.0
+      with:
+        version-source: file
+        version-file: VERSION
+        release-version-cut-prerelease: true
+        next-version-increment-patch: true
+        next-version-increment-minor: false
+        next-version-increment-major: false
+
+    #Increase Version in Build.gradle file
+    - name: Increase version code and change version name
+      shell: bash
+      run: |
+        echo "${{ env.RELEASE_VERSION }}" > VERSION
+        current_version_code=$(cat VERSION_CODE)
+        echo Version Code: $current_version_code
+        echo Version Name: ${{ env.RELEASE_VERSION }}
+
+        sed -i "s/versionCode [0-9]*/versionCode $current_version_code/g" app/build.gradle
+        sed -i "s/versionName \"[^\"]*\"/versionName \"${{ env.RELEASE_VERSION }}\"/g" app/build.gradle
+        current_version_code=$((current_version_code+1))
+        echo "$current_version_code" > VERSION_CODE
+        echo "${{ env.NEXT_VERSION }}" > VERSION
+   
+      #JDK 1.8
+    - name: Set up JDK 1.8
```

is this supposed to be?
```suggestion
    - name: Set up JDK 18
```